### ISSUE TITLE
Suggestion: attachment ease of use

### ISSUE THREAD

**Issue by a7hybnj2** [inventree/InvenTree#614]
Evertything makes sense. I thought today that it would be nice to associate the datasheet to a part and behold there is already an attachments area. Perfect.

My suggestion applys to both attaching picture, images, and whatever may be attached via the attachment areas. Allow pulling via hyperlink. Ex:

If I need a datasheet for a part say a [resistor](https://www.mouser.com/ProductDetail/Panasonic/ERJ-6GEY0R00V?qs=sGAEpiMZZMvdGkrng054tw5%2FFYq5P%2FDotJQTo51%252B1ig%3D) it seems like several extra steps to download the [pdf](https://www.mouser.com/datasheet/2/315/AOA0000C301-1488782.pdf), save the pdf, attach the pdf to InvenTree, then delete the local copy of the pdf.

It would be amazing if we could paste the link to the [pdf](https://www.mouser.com/datasheet/2/315/AOA0000C301-1488782.pdf) into the attachment area and have it automatically download and link it.

**Comment by SchrodingersGat** [comment]
I actually started looking into the pull-via-hyperlink approach (e.g. drag-and-drop an image or link from a browser).

However, two major issues:

1. The javascript required to do this is *really* complicated and seems to be vastly different per browser. And there's no guarantee it would remain supported.
1. It (IMO) could introduce security issues - at the very least this needs to be carefully considered

I think that the addition of drag-and-drop support for file attachment would be really useful, and relatively easy to achieve. The hyperlink stuff requires some further thought - and at this stage would require too much dev time which I think is better spent elsewhere (e.g. key functionality).

**Comment by a7hybnj2** [comment]
LTS is important.

As I am adding parts to my own inventory from suppliers I am often just borrowing their pictures for the part. I have now come to realize they do not have a unique picture for everything. EX: 0805 Thick File Resistors will mainly use just one image. Sometimes there is a unique image per manufacture. Anyway, have you thought about that? Linking 1 image to multiple parts?

**Comment by a7hybnj2** [comment]
@SchrodingersGat Hey, I found a few sites like imgur, github, gitlab, ect. that have drag and drop capabilities. On all of these sites I can drag an image from one webpage and drop it into the drag and drop area... I don't know what that means but I was excited. If you did drag and drop maybe it would just magically allow drag and drop from other sites.
This is my test image: https://www.mouser.com/images/kemet/hd/SMD_MLCC_new_t.jpg
and I just draged it below
```
![](https://www.mouser.com/images/kemet/hd/SMD_MLCC_new_t.jpg)
```
![](https://www.mouser.com/images/kemet/hd/SMD_MLCC_new_t.jpg)

**Comment by eeintech** [comment]
@a7hybnj2 I suggest you take a look at InvenTree's python interface: https://github.com/inventree/inventree-python

With a few lines of Python code you can download a picture/file from the supplier's website (wget for instance) and upload it to a specific part on InvenTree (Thumbnail or attachment, you have the options). If you have any issue or question, you can post them here: https://github.com/inventree/inventree-python/issues

**Comment by ivladz** [comment]
This is basically the same request as #1306 
I was thinking of an easier method: copy-paste the hyperlink instead of drag-and-drop. @a7hybnj2 do you think that this approach would work for you?

**Comment by a7hybnj2** [comment]
@ivladz That would certainly be an improvement. I was thinking about it from the perspective of not just pulling images from websites. I will often take my own picture of the inventory which help identify it better than some of the generic pictures you find for parts online. I take a batch of photos then view my pictures in thumbnail view to be able to select the correct one. In certain application/websites (like github) you can drag and drop. Like ⇩when your making a comment (that arrow is pointing to the box that you drag them into).

**Comment by ivladz** [comment]
@a7hybnj2 Good point. I think overall several quality of life changes are needed for images/attachments:
- Adding via hyperlink (copy paste method)
- Drag and drop local file
- Take photo from camera directly from webpage
- Add photo directly from phone APP (gallery or new from camera)

In short, any one of them brings something to the table.

**Comment by a7hybnj2** [comment]
@ivladz I concur. Also the order you have listed seems like a good order of implementation because it seems to also be easiest to most difficult.￼￼
